crates/tap-agent/src/tracker/generic_tracker.rs

```diff
@@ -214,6 +214,13 @@ where
         });
     }
 
+    pub fn is_allocation_id_blocked(&self, address: &Address) -> bool {
+        self.id_to_fee
+            .get(address)
+            .map(|v| v.blocked)
+            .expect("Allocation ID not found")
```

Comment:
I get it's probably unlikely but maybe let's handle the address not being found with a `match` or something and log it so we don't panic.